lti_consumer/lti_1p3/extensions/rest_framework/pagination.py

```diff
@@ -0,0 +1,158 @@
+"""
+Custom Pagination Implementation
+"""
+from rest_framework.pagination import PageNumberPagination
+from rest_framework.response import Response
+from lti_consumer.lti_1p3.exceptions import LtiNRPSEnrollmentLimitExceed
+from lti_consumer.lti_1p3.constants import LTI_1P3_CONTEXT_ROLE_MAP
+from lti_consumer.plugin.compat import get_course_members
+from lti_consumer.utils import lti_nrps_enrollment_limit
+
+
+class LinkHeaderPagination(PageNumberPagination):
+    """
+    Pagination via link header.
+    """
+
+    page_size_query_param = "limit"
+
+    def get_paginated_response(self, data):
+        """
+        Add next and prev page urls to Link header if exists.
+        """
+        next_page = self.get_next_link()
+        prev_page = self.get_previous_link()
+
+        links = [
+            (next_page, 'next'),
+            (prev_page, 'prev'),
+        ]
+
+        header_links = ['<{}>; rel="{}"'.format(link, rel) for link, rel in links if link]
+
+        headers = {'Link': ', '.join(header_links)} if len(header_links) > 0 else {}
+
+        return Response(data, headers=headers)
+
+
+class LTINRPSMembershipPage:
+    """
+    This class mimics django paginator page class to support
+    the output of `get_course_membership` LMS API.
+    """
+    def __init__(self, data):
+        self.data = data
+
+    def has_next(self):
+        """
+        Checks if there is a next page.
+        """
+        return int(self.data['current_page']) < int(self.data['num_pages'])
+
+    def has_previous(self):
+        """
+        Checks if there is a previous page.
+        """
+        return int(self.data['current_page']) > 1
+
+    def next_page_number(self):
+        """
+        Get next page number
+        """
+        return int(self.data['current_page']) + 1
+
+    def previous_page_number(self):
+        """
+        Get previous page number
+        """
+        return int(self.data['current_page']) - 1
+
+
+class LTINRPSMembersipPagination(LinkHeaderPagination):
+    """
+    This class helps DRF to paginate over the output of `get_course_membership` LMS API.
+    """
+
+    def parse_role_filter(self):
+        """
+        Parse role filter from query param and map it to edx role.
+        """
+        role_filter_param = self.request.query_params.get('role', None)
+
+        if role_filter_param:
+            for edx_role, context_roles in LTI_1P3_CONTEXT_ROLE_MAP.items():
+                if role_filter_param in context_roles:
+                    return edx_role
+
+        return None
```

Comment:
This is the implicit return value. You can keep it here only if you add an explanation.
```suggestion

        # In case no filters match, bla bla....
        return None
```

Reply:
Added explanation.